### PR TITLE
Branch fix - Github API rate limit exceeded

### PR DESCRIPTION
**Pull request by keyonghan**
This PR fixes https://github.com/flutter/flutter/issues/54293

This is a temporary fix. Will create a follow up PR to save branch list info in datastore, and retrieve it from cocoon config. This should get rid of github API rate limit.

### CODE REVIEW

File: app_dart/lib/src/foundation/utils.dart [hunk 1]
```diff
@@ -68,6 +68,39 @@ Future<List<String>> loadBranchRegExps(
   return <String>['master'];
 }
 
+Future<List<String>> getBranchList(
+    Config config,
+    HttpClientProvider branchHttpClientProvider,
+    Logging log,
+    GitHubBackoffCalculator gitHubBackoffCalculator) async {
+  final List<String> regExps = await loadBranchRegExps(
+      branchHttpClientProvider, log, gitHubBackoffCalculator);
```
Comment (godofredoc) [inline app_dart/lib/src/foundation/utils.dart:77]: Is this a temporary fix? we probably should avoid hardcoding branch names.

Comment (godofredoc) [reply app_dart/lib/src/foundation/utils.dart:77]: If it is a temporary fix please add a todo with the bug number to remove it.

Comment (keyonghan) [reply app_dart/lib/src/foundation/utils.dart:77]: Added that in this PR description. I will have a follow up PR soon.

Comment (godofredoc) [reply app_dart/lib/src/foundation/utils.dart:77]: I think is better to add todos and bug number to the code. Is difficult to keep track of them from PR descriptions.

Comment (keyonghan) [reply app_dart/lib/src/foundation/utils.dart:77]: Yeah. Just added: https://github.com/flutter/flutter/issues/54297

**Pull request merged**